domain/application/service/application.go

```diff
@@ -1289,7 +1303,15 @@ func (s *Service) SetApplicationStatus(
 		return internalerrors.Errorf("encoding workload status: %w", err)
 	}
 
-	return s.st.SetApplicationStatus(ctx, applicationID, encodedStatus)
+	if err := s.st.SetApplicationStatus(ctx, applicationID, encodedStatus); err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	if status == nil {
+		return nil
+	}
```

Comment:
Can we move this to the beginning of the method to return early? Do we need to to call `SetApplicationStatus` even if it's nil?